src/dashentry.rs

```diff
@@ -0,0 +1,206 @@
+use std::{hash::BuildHasher, marker::PhantomData};
+
+use crate::typedvalue::SyncTypedMapValue;
+use crate::TypedMapKey;
+use crate::{dashmap::RefMut, typedkey::SyncTypedKey};
+
+pub struct OccupiedEntry<'a, K, Marker, S>
+where
+    K: 'static + TypedMapKey<Marker>,
```

Comment:
Requireing keys seems restrictive. Would it prevent me from doing something like?:
``` rust
let my_string = "foo".to_owned();
my_map.insert(my_string.as_str(), 42);
```

Reply:
Can it be K: 'a + Typed...?

Reply:
Probably not. Tried to work around it a bit but without success. In the end downcast can be done on Box<dyn Any + 'static>. Requiring static probably makes sense anyway, because when downcasting you may not know original lifetime.